.github/workflows/ci.yml

```diff
@@ -0,0 +1,106 @@
+---
+
+name: ci
+
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+
+# Limit the access of the generated GITHUB_TOKEN
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    permissions:
+      statuses: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+      - run: '.ci/scripts/lint.sh'
+      - run: 'npm run ci:bundlesize'
+        id: bundlesize
+      - uses: actions/github-script@v6
+        env:
+          BUNDLESIZE_PASS: ${{ steps.bundlesize.outputs.bundlesize_pass }}
+          BUNDLESIZE_FAIL: ${{ steps.bundlesize.outputs.bundlesize_fail }}
+        with:
+          script: |
+            const { BUNDLESIZE_PASS, BUNDLESIZE_FAIL } = process.env;
+            let state = 'success';
+            if (BUNDLESIZE_FAIL > 0) {
+              state = 'failure';
+            }
+            github.rest.repos.createCommitStatus({
+              ...context.repo,
+              sha: context.sha,
+              state: state,
+              context: `${context.workflow} / Bundlesize: ${BUNDLESIZE_PASS} pass / ${BUNDLESIZE_FAIL} fail`,
+            });
+
+  test-puppeteer:
+    name: Test Puppeteer
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        stack-version:
+          - '8.0.0-SNAPSHOT'
+          - '8.6.1'
+        scope:
+          - '@elastic/apm-rum'
+          - '@elastic/apm-rum-core'
+          - '@elastic/apm-rum-react'
+          - '@elastic/apm-rum-angular'
+          - '@elastic/apm-rum-vue'
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run puppeteer tests
+        uses: ./.github/workflows/run-test
+        with:
+          goal: 'test'
+          scope: ${{ matrix.scope }}
+          stack-version: ${{ matrix.stack-version }}
+
+  all-tests-puppeteer:
+    name: All Tests Puppeteer
+    if: always()
+    runs-on: ubuntu-latest
+    needs:
+      - lint
+      - test-puppeteer
+      - coverage
+    steps:
+      - id: check
+        uses: elastic/apm-pipeline-library/.github/actions/check-dependent-jobs@current
+        with:
+          needs: ${{ toJSON(needs) }}
+      - run: ${{ steps.check.outputs.isSuccess }}
+
+  coverage:
```

Comment:
minor NIT: since this job runs in parallel and it's needed by the `all-tests-puppeteer`, maybe it could be moved above so `all-tests-puppeteer` will be the last job since it needs the other ones?

Reply:
I removed the need for lint and coverage. We only need to wait for matrix jobs to generate the right check.